modules/Bio/EnsEMBL/Compara/RunnableDB/Ortheus.pm

```diff
@@ -438,63 +540,53 @@ sub parse_results {
     pop(@$ids); ## ...except for the last leaf which is the end of the tree
     #print join(" :: ", @$ids), "\n\n";
 
-    while (<$fh>) {
-	next if (/^\s*$/);
-	chomp;
-	## FASTA headers correspond to the tree and the order of the leaves in the tree corresponds
-	## to the order of the files
-
-	if (/^>/) {
-	    print "PARSING $_\n" if ($self->debug);
-	    my ($name) = $_ =~ /^>(.+)/;
-	    if (defined($this_genomic_align) and  $seq) {
-		    print "add aligned_sequence " . $this_genomic_align->dnafrag_id . " " . $this_genomic_align->dnafrag_start . " " . $this_genomic_align->dnafrag_end . "\n" if $self->debug;
+    my $parser = Bio::EnsEMBL::IO::Parser::Fasta->open($alignment_file);
+    while ($parser->next()) {
+        my $name = $parser->getHeader;
+        my $seq  = $parser->getSequence;
 
+		    my $this_genomic_align = new Bio::EnsEMBL::Compara::GenomicAlign;
 		    $this_genomic_align->aligned_sequence($seq);
 		    $this_genomic_align_block->add_GenomicAlign($this_genomic_align);
-	    }
-	    my $header = shift(@$ids);
-	    $this_genomic_align = new Bio::EnsEMBL::Compara::GenomicAlign;
+		    my $genomic_align_group = new Bio::EnsEMBL::Compara::GenomicAlignGroup;
+		    $genomic_align_group->add_GenomicAlign($this_genomic_align);
 
-	    if (!defined($header)) {
-		print "INTERNAL NODE $name\n" if ($self->debug);
 		my $this_node;
 		foreach my $this_leaf_name (split("_", $name)) {
+		    my $this_leaf = $tree->find_node_by_name($this_leaf_name);
+		    if (!$this_leaf) {
+			die "Unable to find_node_by_name $this_leaf_name";
+		    }
 		    if ($this_node) {
-			my $other_node = $tree->find_node_by_name($this_leaf_name);
-			if (!$other_node) {
-			    throw("Cannot find node <$this_leaf_name>\n");
-			}
-			$this_node = $this_node->find_first_shared_ancestor($other_node);
+			$this_node = $this_node->find_first_shared_ancestor($this_leaf);
 		    } else {
-			print "LEAF: $this_leaf_name\n" if ($self->debug);
-			$this_node = $tree->find_node_by_name($this_leaf_name);
+			$this_node = $this_leaf;
 		    }
 		}
-		print join("_", map {$_->name} @{$this_node->get_all_leaves}), "\n" if ($self->debug);
+		print "Found node: ", join("_", map {$_->name} @{$this_node->get_all_leaves}), "\n" if ($self->debug);
```

Comment:
```suggestion
        print "Found node: ", join("_", map {$_->name} @{$this_node->get_all_leaves}), "\n" if ($self->{debug});
```
I don't remember the details, but I remember having an issue with `$self->debug` in the past, but the hash access was fine. Maybe I messed something else :man_shrugging:

Reply:
Unexpected as it's not stored in `$self` directly, cf https://github.com/Ensembl/ensembl-hive/blob/version/2.5/modules/Bio/EnsEMBL/Hive/Process.pm#L641-L654